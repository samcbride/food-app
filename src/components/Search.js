/* eslint-disable no-lone-blocks */
import React, { useState } from "react";
import PropTypes from "prop-types";
import fetchRecipes from "../requests/axiosrecipe";

const Search = ({ handleSubmit }) => {
  
  const [inputValue, setInputValue] = useState();

  const handleSearch = async (event) => {
    event.preventDefault();
    handleSubmit(await fetchRecipes(inputValue));
  };

  const handleInputChange = (event) => {
    setInputValue({ ...inputValue, [event.target.name]: event.target.value });
  };

  return (
    <form onSubmit={handleSearch}>
      <label>
        Enter your ingredients:
        <input
          placeholder="Search"
          type="text"
          id="ingredient"
          name="ingredient"
          //value={initialState.ingredient}
          onChange={handleInputChange}
        />
        {/* <option value="tomato">Tomato</option>
          <option value="pasta">Pasta</option>
          <option value="basil">Basil</option>
        </select> */}
      </label>
      <button type="submit">Add</button>
    </form>
  );
};

export default Search;

Search.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
};

// npm i --save react-select
// import Select from "react-select";
{
  /* //   const initialState = [
  //     { label: "tomato", value: "tomato" },
  //     { label: "pasta", value: "pasta" },
  //     { label: "basil", value: "basil" },
  //     { label: "parmesan", value: "parmesan" },
  //   ]; */
}
{
  /* <select
          type="text"
          name="search"
          placeholder="Ingredients"
          options={initialState}
          onChange={handleInputChange}
        ></select> */
}
