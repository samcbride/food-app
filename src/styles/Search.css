.search {
  position: relative;
  display: flex;
  min-width: 100px;
  justify-content: right;
}

.input {
  border: 1px solid lightgray;
  width: 90%;
}
button {
  border: 1px solid transparent;
  transition-duration: 0.4s;
  cursor: pointer;
  color: #008cba;
  background: honeydew;
}
button:hover {
  background-color: #83aec3;
  color: honeydew;
  box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
}

.input,
.button {
  padding: 10px 15px;
  font-size: 18px;
  border-radius: 5px;
  font-family: "Amaranth", sans-serif;
}

.error {
  padding-top: 0.25em;
  margin: 0.5em;
}

@media (max-width: 530px) {
  .search {
    margin: 1em;
  }
}
