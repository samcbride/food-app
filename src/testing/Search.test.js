import { render } from "@testing-library/react";
import Search from "../components/Search";

describe("Search", () => {
  const validProps = {
    handleSubmit: () => {},
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <Search handleSubmit={validProps.handleSubmit} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("check button exists with correct text", () => {
    const { getByRole } = render(
      <Search handleSubmit={validProps.handleSubmit} />
    );
    const button = getByRole("button", { name: /Search/i });
    expect(button).toBeInTheDocument();
  });
});
