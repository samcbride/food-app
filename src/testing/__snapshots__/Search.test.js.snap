// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<DocumentFragment>
  <form>
    <div
      class="search"
    >
      <label>
        <input
          class="input"
          id="ingredient"
          name="ingredient"
          placeholder="Enter your ingredients"
          type="text"
        />
      </label>
      <button
        class="search-button"
        type="submit"
      >
        Search
      </button>
    </div>
  </form>
</DocumentFragment>
`;
